From 469ce67bbf3156091133e69daee9af7f9fb70102 Mon Sep 17 00:00:00 2001
From: BenjiReis <benjamin.reis@vates.fr>
Date: Mon, 25 Sep 2023 16:13:13 +0200
Subject: [PATCH 23/25] Support IPv6 in Ceph Driver

Signed-off-by: BenjiReis <benjamin.reis@vates.fr>
---
 drivers/CephFSSR.py | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/drivers/CephFSSR.py b/drivers/CephFSSR.py
index d85faf9..2f8c50d 100644
--- a/drivers/CephFSSR.py
+++ b/drivers/CephFSSR.py
@@ -20,6 +20,7 @@
 
 import errno
 import os
+import socket
 import syslog as _syslog
 import xmlrpc.client
 from syslog import syslog
@@ -141,7 +142,17 @@ class CephFSSR(FileSR.FileSR):
                 options.append(self.dconf['options'])
             if options:
                 options = ['-o', ','.join(options)]
-            command = ["mount", '-t', 'ceph', self.remoteserver+":"+self.remoteport+":"+self.remotepath, mountpoint] + options
+            acc = []
+            for server in self.remoteserver.split(','):
+                try:
+                    addr_info = socket.getaddrinfo(server, 0)[0]
+                except Exception:
+                    continue
+
+                acc.append('[' + server + ']' if addr_info[0] == socket.AF_INET6 else server)
+
+            remoteserver = ','.join(acc)
+            command = ["mount", '-t', 'ceph', remoteserver + ":" + self.remoteport + ":" + self.remotepath, mountpoint] + options
             util.ioretry(lambda: util.pread(command), errlist=[errno.EPIPE, errno.EIO], maxretry=2, nofail=True)
         except util.CommandException as inst:
             syslog(_syslog.LOG_ERR, 'CephFS mount failed ' + inst.__str__())
-- 
2.45.2

