From 006a3727a18eed16a581251553b829667efceefd Mon Sep 17 00:00:00 2001
From: Mark Syms <mark.syms@cloud.com>
Date: Fri, 30 Aug 2024 10:13:27 +0100
Subject: [PATCH 28/28] CA-398425: correctly check for multiple targets in
 iSCSI

Signed-off-by: Mark Syms <mark.syms@cloud.com>
---
 drivers/LVHDoISCSISR.py | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/LVHDoISCSISR.py b/drivers/LVHDoISCSISR.py
index cbde1b9..442ec30 100755
--- a/drivers/LVHDoISCSISR.py
+++ b/drivers/LVHDoISCSISR.py
@@ -103,7 +103,8 @@ class LVHDoISCSISR(LVHDSR.LVHDSR):
         self.iscsiSRs = []
         self.iscsiSRs.append(iscsi)
         saved_exc = None
-        if self.dconf['target'].find(',') == 0 or self.dconf['targetIQN'] == "*":
+        targets = self.dconf['target'].split(',')
+        if len(targets) > 1 or self.dconf['targetIQN'] == "*":
             # Instantiate multiple sessions
             self.iscsiSRs = []
             if self.dconf['targetIQN'] == "*":
@@ -129,7 +130,7 @@ class LVHDoISCSISR(LVHDSR.LVHDSR):
                     # IQNstring cannot be built with a dictionary iteration because of this
                     IQNstring = self.dconf['multiSession']
                 else:
-                    for tgt in self.dconf['target'].split(','):
+                    for tgt in targets:
                         try:
                             tgt_ip = util._convertDNS(tgt)
                         except:
