From dfd6645bb4b8d66b2c613cad144c5c84f3ce5fcc Mon Sep 17 00:00:00 2001
From: Ronan Abhamon <ronan.abhamon@vates.tech>
Date: Thu, 10 Apr 2025 10:39:54 +0200
Subject: [PATCH] fix(LinstorSR): simplify _kick_gc code using systemd service

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.tech>
---
 drivers/LinstorSR.py | 24 ++----------------------
 1 file changed, 2 insertions(+), 22 deletions(-)

diff --git a/drivers/LinstorSR.py b/drivers/LinstorSR.py
index 829c48f8..1ff3d963 100755
--- a/drivers/LinstorSR.py
+++ b/drivers/LinstorSR.py
@@ -31,7 +31,7 @@ except ImportError:
 
     LINSTOR_AVAILABLE = False
 
-from lock import Lock, LOCK_TYPE_GC_RUNNING
+from lock import Lock
 import blktap2
 import cleanup
 import distutils
@@ -1516,28 +1516,8 @@ class LinstorSR(SR.SR):
             raise xs_errors.XenError('SRNoSpace')
 
     def _kick_gc(self):
-        # Don't bother if an instance already running. This is just an
-        # optimization to reduce the overhead of forking a new process if we
-        # don't have to, but the process will check the lock anyways.
-        lock = Lock(LOCK_TYPE_GC_RUNNING, self.uuid)
-        if not lock.acquireNoblock():
-            if not cleanup.should_preempt(self.session, self.uuid):
-                util.SMlog('A GC instance already running, not kicking')
-                return
-
-            util.SMlog('Aborting currently-running coalesce of garbage VDI')
-            try:
-                if not cleanup.abort(self.uuid, soft=True):
-                    util.SMlog('The GC has already been scheduled to re-start')
-            except util.CommandException as e:
-                if e.code != errno.ETIMEDOUT:
-                    raise
-                util.SMlog('Failed to abort the GC')
-        else:
-            lock.release()
-
         util.SMlog('Kicking GC')
-        cleanup.gc(self.session, self.uuid, True)
+        cleanup.start_gc_service(self.uuid)
 
 # ==============================================================================
 # LinstorSr VDI
