From ea2f0b4686cd039785a753e0040ff1b3be20d5d1 Mon Sep 17 00:00:00 2001
From: Ronan Abhamon <ronan.abhamon@vates.fr>
Date: Thu, 10 Oct 2024 14:58:02 +0200
Subject: [PATCH] Use importlib instead of imp which is deprecated in python
 3.4

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>
---
 drivers/SR.py | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/SR.py b/drivers/SR.py
index 9ca0045c..d1aacdb3 100755
--- a/drivers/SR.py
+++ b/drivers/SR.py
@@ -153,7 +153,7 @@ class SR(object):
 
     @staticmethod
     def from_uuid(session, sr_uuid):
-        import imp
+        import importlib.util
 
         _SR = session.xenapi.SR
         sr_ref = _SR.get_by_uuid(sr_uuid)
@@ -169,7 +169,10 @@ class SR(object):
         driver_real = os.path.realpath(driver_path)
         module_name = os.path.basename(driver_path)
 
-        module = imp.load_source(module_name, driver_real)
+        spec = importlib.util.spec_from_file_location(module_name, driver_real)
+        module = importlib.util.module_from_spec(spec)
+        spec.loader.exec_module(module)
+
         target = driver(sm_type)
         # NB. get the host pbd's device_config
 
