From 49fad6ce14acee828ef7d5c90abbc1a4310b788c Mon Sep 17 00:00:00 2001
From: Ronan Abhamon <ronan.abhamon@vates.fr>
Date: Fri, 13 May 2022 14:35:57 +0200
Subject: [PATCH 061/178] fix(minidrbdcluster): supports new properties like
 `force-io-failures`

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>
---
 scripts/minidrbdcluster | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/scripts/minidrbdcluster b/scripts/minidrbdcluster
index a04b6c1..fb4de09 100755
--- a/scripts/minidrbdcluster
+++ b/scripts/minidrbdcluster
@@ -11,15 +11,15 @@ DRBDADM_OPEN_FAILED_RE = re.compile(
 )
 MAY_PROMOT_RE = re.compile(
     '(?:exists|change) resource name:((?:\\w|-)+) '
-    '(?:\\w+\\:\\w+ )*may_promote:(yes|no) promotion_score:(\\d+)'
+    '(?:(?:\\w|-)+\\:(?:\\w|-)+ )*may_promote:(yes|no) promotion_score:(\\d+)'
 )
 PEER_ROLE_RE = re.compile(
     '(?:exists|change) connection name:((?:\\w|-)+) peer-node-id:(?:\\d+) '
-    'conn-name:(\\w+) (?:\\w+\\:\\w+ )*role:(Primary|Secondary|Unknown)'
+    'conn-name:((?:\\w|-)+) (?:(?:\\w|-)+\\:(?:\\w|-)+ )*role:(Primary|Secondary|Unknown)'
 )
 HAVE_QUORUM_RE = re.compile(
     '(?:exists|change) device name:((?:\\w|-)+) '
-    '(?:\\w+\\:\\w+ )*quorum:(yes|no)'
+    '(?:(?:\\w|-)+\\:(?:\\w|-)+ )*quorum:(yes|no)'
 )
 
 
-- 
2.46.0

