From 97ef3f8ae7ecb992a2abea344e9d53ff4f3ae11a Mon Sep 17 00:00:00 2001
From: Damien Thenot <damien.thenot@vates.tech>
Date: Thu, 9 Jan 2025 17:41:32 +0100
Subject: [PATCH] Fix filter to reject other device types (#77)

Signed-off-by: Damien Thenot <damien.thenot@vates.tech>
---
 drivers/LargeBlockSR.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/LargeBlockSR.py b/drivers/LargeBlockSR.py
index fa66cfce..449b2cf1 100644
--- a/drivers/LargeBlockSR.py
+++ b/drivers/LargeBlockSR.py
@@ -224,7 +224,7 @@ class LargeBlockSR(EXTSR.EXTSR):
         util.SMlog("Reconnecting VG {} to use emulated device".format(self.vgname))
         try:
             lvutil.setActiveVG(self.vgname, False)
-            lvutil.setActiveVG(self.vgname, True, config="devices{ global_filter = [ \"r|^/dev/nvme.*|\", \"a|/dev/loop.*|\" ] }")
+            lvutil.setActiveVG(self.vgname, True, config="devices{ global_filter = [ \"a|/dev/loop.*|\", \"r|.*|\" ] }")
         except util.CommandException as e:
             xs_errors.XenError("LargeBlockVGReconnectFailed", opterr="Failed to reconnect the VolumeGroup {}, error: {}".format(self.vgname, e))
 
