From 01433213562b35e8d9f9d20b04105af7fd570bb3 Mon Sep 17 00:00:00 2001
From: Ronan Abhamon <ronan.abhamon@vates.fr>
Date: Sat, 14 Dec 2024 00:28:06 +0100
Subject: [PATCH] feat(VdiType): define QCOW2 type

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>
---
 drivers/vditype.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/vditype.py b/drivers/vditype.py
index 159f6c2b..3a478c7a 100644
--- a/drivers/vditype.py
+++ b/drivers/vditype.py
@@ -20,6 +20,7 @@ from sm_typing import Final
 class VdiType(object):
     RAW = "aio"
     VHD = "vhd"
+    QCOW2 = "qcow2"
     ISO = "iso"
     FILE = "file"
     CBTLOG = "cbtlog"
@@ -32,6 +33,7 @@ class VdiType(object):
 class VdiTypeExtension(object):
     RAW = ".raw"
     VHD = ".vhd"
+    QCOW2 = ".qcow2"
     ISO = ".iso"
     FILE = ".file"
     CBTLOG = ".cbtlog"
@@ -39,6 +41,7 @@ class VdiTypeExtension(object):
 VDI_TYPE_TO_EXTENSION: Final = {
     VdiType.RAW: VdiTypeExtension.RAW,
     VdiType.VHD: VdiTypeExtension.VHD,
+    VdiType.QCOW2: VdiTypeExtension.QCOW2,
     VdiType.ISO: VdiTypeExtension.ISO,
     VdiType.FILE: VdiTypeExtension.FILE,
     VdiType.CBTLOG: VdiTypeExtension.CBTLOG
