From 1f3b535546aab6f7f0839450fb4aed28253d580b Mon Sep 17 00:00:00 2001
From: Ronan Abhamon <ronan.abhamon@vates.fr>
Date: Thu, 19 Dec 2024 15:19:15 +0100
Subject: [PATCH] refactor(cleanup.py): rename DB_VHD_PARENT to DB_VDI_PARENT

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>
---
 drivers/cleanup.py | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/drivers/cleanup.py b/drivers/cleanup.py
index b4ef5ce3..75e48d75 100755
--- a/drivers/cleanup.py
+++ b/drivers/cleanup.py
@@ -505,7 +505,7 @@ class VDI(object):
     DEVICE_MAJOR = 202
 
     # config keys & values
-    DB_VHD_PARENT = "vhd-parent"
+    DB_VDI_PARENT = "vhd-parent"
     DB_VDI_TYPE = "vdi_type"
     DB_VHD_BLOCKS = "vhd-blocks"
     DB_VDI_PAUSED = "paused"
@@ -526,7 +526,7 @@ class VDI(object):
     DB_ALLOW_CACHING = "allow_caching"
 
     CONFIG_TYPE = {
-            DB_VHD_PARENT: XAPI.CONFIG_SM,
+            DB_VDI_PARENT: XAPI.CONFIG_SM,
             DB_VDI_TYPE: XAPI.CONFIG_SM,
             DB_VHD_BLOCKS: XAPI.CONFIG_SM,
             DB_VDI_PAUSED: XAPI.CONFIG_SM,
@@ -1048,7 +1048,7 @@ class VDI(object):
         self.parentUuid = parent.uuid
         parent.children.append(self)
         try:
-            self.setConfig(self.DB_VHD_PARENT, self.parentUuid)
+            self.setConfig(self.DB_VDI_PARENT, self.parentUuid)
             Util.log("Updated the vhd-parent field for child %s with %s" % \
                      (self.uuid, self.parentUuid))
         except:
@@ -1437,7 +1437,7 @@ class LVHDVDI(VDI):
         self.parentUuid = parent.uuid
         parent.children.append(self)
         try:
-            self.setConfig(self.DB_VHD_PARENT, self.parentUuid)
+            self.setConfig(self.DB_VDI_PARENT, self.parentUuid)
             Util.log("Updated the vhd-parent field for child %s with %s" % \
                      (self.uuid, self.parentUuid))
         except:
@@ -1704,7 +1704,7 @@ class LinstorVDI(VDI):
         self.parentUuid = parent.uuid
         parent.children.append(self)
         try:
-            self.setConfig(self.DB_VHD_PARENT, self.parentUuid)
+            self.setConfig(self.DB_VDI_PARENT, self.parentUuid)
             Util.log("Updated the vhd-parent field for child %s with %s" % \
                      (self.uuid, self.parentUuid))
         except:
@@ -2693,7 +2693,7 @@ class SR(object):
         # garbage
 
         # update the VDI record
-        vdi.parent.delConfig(VDI.DB_VHD_PARENT)
+        vdi.parent.delConfig(VDI.DB_VDI_PARENT)
         if vdi.parent.vdi_type == VdiType.RAW:
             vdi.parent.setConfig(VDI.DB_VDI_TYPE, VdiType.RAW)
         vdi.parent.delConfig(VDI.DB_VHD_BLOCKS)
@@ -2997,7 +2997,7 @@ class FileSR(SR):
             Util.log("Renaming child back to %s" % childUuid)
             child.rename(childUuid)
             Util.log("Updating the VDI record")
-            child.setConfig(VDI.DB_VHD_PARENT, parentUuid)
+            child.setConfig(VDI.DB_VDI_PARENT, parentUuid)
             child.setConfig(VDI.DB_VDI_TYPE, VdiType.VHD)
             util.fistpoint.activate("LVHDRT_coaleaf_undo_after_rename2", self.uuid)
 
@@ -3217,7 +3217,7 @@ class LVHDSR(SR):
             Util.log("Renaming child back to %s" % childUuid)
             child.rename(childUuid)
             Util.log("Updating the VDI record")
-            child.setConfig(VDI.DB_VHD_PARENT, parentUuid)
+            child.setConfig(VDI.DB_VDI_PARENT, parentUuid)
             child.setConfig(VDI.DB_VDI_TYPE, VdiType.VHD)
             util.fistpoint.activate("LVHDRT_coaleaf_undo_after_rename2", self.uuid)
 
@@ -3585,7 +3585,7 @@ class LinstorSR(SR):
             Util.log('Renaming child back to {}'.format(childUuid))
             child.rename(childUuid)
             Util.log('Updating the VDI record')
-            child.setConfig(VDI.DB_VHD_PARENT, parentUuid)
+            child.setConfig(VDI.DB_VDI_PARENT, parentUuid)
             child.setConfig(VDI.DB_VDI_TYPE, VdiType.VHD)
 
         # TODO: Maybe deflate here.
