From 7928957e8d34e46b990d58da2f9264b4984da5cd Mon Sep 17 00:00:00 2001
From: Ronan Abhamon <ronan.abhamon@vates.fr>
Date: Wed, 14 Sep 2022 10:17:18 +0200
Subject: [PATCH 077/177] fix(LinstorSR): disable VHD key hash usage to limit
 exec time

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>
---
 drivers/LinstorSR.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/LinstorSR.py b/drivers/LinstorSR.py
index 47ac3c85..374d6cb9 100755
--- a/drivers/LinstorSR.py
+++ b/drivers/LinstorSR.py
@@ -72,6 +72,9 @@ USE_HTTP_NBD_SERVERS = True
 # Useful flag to trace calls using cProfile.
 TRACE_PERFS = False
 
+# Enable/Disable VHD key hash support.
+USE_KEY_HASH = False
+
 # ==============================================================================
 
 # TODO: Supports 'VDI_INTRODUCE', 'VDI_RESET_ON_BOOT/2', 'SR_TRIM',
@@ -1234,7 +1237,7 @@ class LinstorSR(SR.SR):
             vdi = self.vdi(vdi_uuid)
             self.vdis[vdi_uuid] = vdi
 
-            if vdi.vdi_type == vhdutil.VDI_TYPE_VHD:
+            if USE_KEY_HASH and vdi.vdi_type == vhdutil.VDI_TYPE_VHD:
                 # TODO: Replace pylint comment with this feature when possible:
                 # https://github.com/PyCQA/pylint/pull/2926
                 vdi.sm_config_override['key_hash'] = \
