From fb6db534d07e28a174ad79f40822ea26d9d6de76 Mon Sep 17 00:00:00 2001
From: Ronan Abhamon <ronan.abhamon@vates.fr>
Date: Fri, 20 Dec 2024 15:33:17 +0100
Subject: [PATCH] fix(cleanup.py): always use "vdi_type" to display info

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>
---
 drivers/cleanup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/cleanup.py b/drivers/cleanup.py
index bb681d5a..d9c9ad43 100755
--- a/drivers/cleanup.py
+++ b/drivers/cleanup.py
@@ -1362,7 +1362,7 @@ class LVHDVDI(VDI):
 
     @override
     def __str__(self) -> str:
-        strType = "VHD"
+        strType = self.vdi_type
         if self.vdi_type == VdiType.RAW:
             strType = "RAW"
         strHidden = ""
