From 1af4b19fa1eb822f74424d1d1990d9fa7a0b05cb Mon Sep 17 00:00:00 2001
From: Ronan Abhamon <ronan.abhamon@vates.fr>
Date: Fri, 20 Dec 2024 15:42:01 +0100
Subject: [PATCH] refactor(cleanup.py): use allocated_size var for
 getAllocatedSize result

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>
---
 drivers/cleanup.py | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/cleanup.py b/drivers/cleanup.py
index 751a2dc8..52ce4b27 100755
--- a/drivers/cleanup.py
+++ b/drivers/cleanup.py
@@ -681,13 +681,13 @@ class VDI(object):
         feasibleSize = False
         allowedDownTime = \
                 self.TIMEOUT_SAFETY_MARGIN * self.LIVE_LEAF_COALESCE_TIMEOUT
-        vhd_size = self.getAllocatedSize()
+        allocated_size = self.getAllocatedSize()
         if speed:
             feasibleSize = \
-                vhd_size // speed < allowedDownTime
+                allocated_size // speed < allowedDownTime
         else:
             feasibleSize = \
-                vhd_size < self.LIVE_LEAF_COALESCE_MAX_SIZE
+                allocated_size < self.LIVE_LEAF_COALESCE_MAX_SIZE
 
         return (feasibleSize or
                 self.getConfig(self.DB_LEAFCLSC) == self.LEAFCLSC_FORCE)
@@ -918,7 +918,7 @@ class VDI(object):
     def _doCoalesceVHD(vdi):
         try:
             startTime = time.time()
-            vhdSize = vdi.getAllocatedSize()
+            allocated_size = vdi.getAllocatedSize()
             coalesced_size = vdi.coalesce()
             endTime = time.time()
             vdi.sr.recordStorageSpeed(startTime, endTime, coalesced_size)
