From 621b7c365ffb47629c11267396381f6a4b7c62ba Mon Sep 17 00:00:00 2001
From: Damien Thenot <damien.thenot@vates.tech>
Date: Thu, 23 Jan 2025 17:17:00 +0100
Subject: [PATCH 89/89] Enable QCOW2 by default on creation for SR

Signed-off-by: Damien Thenot <damien.thenot@vates.tech>
---
 drivers/SR.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/SR.py b/drivers/SR.py
index 9aad2e98..dff1a11c 100755
--- a/drivers/SR.py
+++ b/drivers/SR.py
@@ -37,7 +37,7 @@ MASTER_LVM_CONF = '/etc/lvm/master'
 # LUN per VDI key for XenCenter
 LUNPERVDI = "LUNperVDI"
 
-DEFAULT_IMAGE_FORMATS = [ImageFormat.VHD, ImageFormat.QCOW2]
+DEFAULT_IMAGE_FORMATS = [ImageFormat.QCOW2, ImageFormat.VHD]
 
 
 
