From 9c0b2f004c602986f61d02c27d387f1749fb09e1 Mon Sep 17 00:00:00 2001
From: Damien Thenot <damien.thenot@vates.tech>
Date: Thu, 23 Jan 2025 17:17:00 +0100
Subject: [PATCH] Enable QCOW2 by default on creation for SR

Signed-off-by: Damien Thenot <damien.thenot@vates.tech>
---
 drivers/SR.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/SR.py b/drivers/SR.py
index 8faee3b4..456de875 100755
--- a/drivers/SR.py
+++ b/drivers/SR.py
@@ -38,7 +38,7 @@ MASTER_LVM_CONF = '/etc/lvm/master'
 # LUN per VDI key for XenCenter
 LUNPERVDI = "LUNperVDI"
 
-DEFAULT_IMAGE_FORMATS = [ImageFormat.VHD, ImageFormat.QCOW2]
+DEFAULT_IMAGE_FORMATS = [ImageFormat.QCOW2, ImageFormat.VHD]
 
 
 
