From ac446e2062cd2d4abcc2084302360151be31c4d1 Mon Sep 17 00:00:00 2001
From: Ronan Abhamon <ronan.abhamon@vates.fr>
Date: Thu, 19 Dec 2024 15:28:50 +0100
Subject: [PATCH] refactor(cleanup.py): rename getVHDBlocks to getVDIBlocks

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>
---
 drivers/cleanup.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/cleanup.py b/drivers/cleanup.py
index d68be6bd..a85df009 100755
--- a/drivers/cleanup.py
+++ b/drivers/cleanup.py
@@ -654,7 +654,7 @@ class VDI(object):
         return util.is_attached_rw(
                 self.sr.xapi.session.xenapi.VDI.get_sm_config(self.getRef()))
 
-    def getVHDBlocks(self):
+    def getVDIBlocks(self):
         val = self.updateBlockInfo()
         bitmap = zlib.decompress(base64.b64decode(val))
         return bitmap
@@ -1119,8 +1119,8 @@ class VDI(object):
         # make sure we don't use stale BAT info from vdi_rec since the child
         # was writable all this time
         self.delConfig(VDI.DB_VDI_BLOCKS)
-        blocksChild = self.getVHDBlocks()
-        blocksParent = self.parent.getVHDBlocks()
+        blocksChild = self.getVDIBlocks()
+        blocksParent = self.parent.getVDIBlocks()
         numBlocks = Util.countBits(blocksChild, blocksParent)
         Util.log("Num combined blocks = %d" % numBlocks)
         sizeData = numBlocks * vhdutil.VHD_BLOCK_SIZE
