From fd8a46e8a1282ce139fab7c41968889c16b26320 Mon Sep 17 00:00:00 2001
From: Ronan Abhamon <ronan.abhamon@vates.tech>
Date: Tue, 22 Apr 2025 11:27:12 +0200
Subject: [PATCH] Fix warns reported by new github workflow

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.tech>
---
 drivers/resetvdis.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/resetvdis.py b/drivers/resetvdis.py
index 0be98e69..607d38f9 100755
--- a/drivers/resetvdis.py
+++ b/drivers/resetvdis.py
@@ -24,7 +24,6 @@ import lock
 import sys
 import XenAPI # pylint: disable=import-error
 
-
 def reset_sr(session, host_uuid, sr_uuid, is_sr_master):
     cleanup.abort(sr_uuid)
 
