From ddf3e56495dc7f1fbb832f9e1b468e4c0cc15c42 Mon Sep 17 00:00:00 2001
From: Damien Thenot <damien.thenot@vates.tech>
Date: Mon, 10 Feb 2025 13:08:35 +0100
Subject: [PATCH] Add function get_openers_pid in util.py

Signed-off-by: Damien Thenot <damien.thenot@vates.tech>
---
 drivers/util.py | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/drivers/util.py b/drivers/util.py
index 825bd306..a4ca0e6a 100755
--- a/drivers/util.py
+++ b/drivers/util.py
@@ -42,6 +42,7 @@ import copy
 import tempfile
 
 from functools import reduce
+from sm_typing import List, Optional
 
 NO_LOGGING_STAMPFILE = '/etc/xensource/no_sm_log'
 
@@ -2072,6 +2073,22 @@ def check_pid_exists(pid):
         return True
 
 
+def get_openers_pid(path: str) -> Optional[List[int]]:
+    cmd = ["lsof", "-t", path]
+
+    try:
+        list = []
+        ret = pread2(cmd)
+        for line in ret.splitlines():
+            list.append(int(line))
+        return list
+    except CommandException as e:
+        if e.code == 1: # `lsof` return 1 if there is no openers
+            return None
+        else:
+            raise e
+
+
 def make_profile(name, function):
     """
     Helper to execute cProfile using unique log file.
