From d6fb1e967d2dcac655b9f5f91095492e810400a8 Mon Sep 17 00:00:00 2001
From: Ronan Abhamon <ronan.abhamon@vates.tech>
Date: Tue, 18 Mar 2025 15:41:16 +0100
Subject: [PATCH] to drop

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.tech>
---
 drivers/cowutil.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/cowutil.py b/drivers/cowutil.py
index 1fb7db0a..19d927b3 100755
--- a/drivers/cowutil.py
+++ b/drivers/cowutil.py
@@ -325,6 +325,10 @@ def getCowUtilFromImageFormat(image_format: ImageFormat) -> CowUtil:
     if image_format in (ImageFormat.RAW, ImageFormat.VHD):
         return vhdutil.VhdUtil()
 
+    # TODO: QCOW2
+
+    return vhdutil.VhdUtil() # TODO: REMOVE
+
     assert False, f"Unsupported image format: {image_format}"
 
 def getCowUtil(vdi_type: str) -> CowUtil:
