From ccdf73530b9858d925cc023d44e880769fe9622b Mon Sep 17 00:00:00 2001
From: Ronan Abhamon <ronan.abhamon@vates.fr>
Date: Fri, 28 Apr 2023 10:43:27 +0200
Subject: [PATCH 119/177] fix(LinstorSR): ensure we always have a DRBD path to
 snap

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>
---
 drivers/LinstorSR.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/LinstorSR.py b/drivers/LinstorSR.py
index 324033a0..8c0b007d 100755
--- a/drivers/LinstorSR.py
+++ b/drivers/LinstorSR.py
@@ -2392,6 +2392,9 @@ class LinstorVDI(VDI.VDI):
         elif depth >= vhdutil.MAX_CHAIN_SIZE:
             raise xs_errors.XenError('SnapshotChainTooLong')
 
+        # Ensure we have a valid path if we don't have a local diskful.
+        self.sr._linstor.get_device_path(self.uuid)
+
         volume_path = self.path
         if not util.pathexists(volume_path):
             raise xs_errors.XenError(
