From 6c31c2abf7a62abc71b455060f2346116d7e9da6 Mon Sep 17 00:00:00 2001
From: Damien Thenot <damien.thenot@vates.tech>
Date: Thu, 23 Jan 2025 17:11:24 +0100
Subject: [PATCH] Add getDefaultPreallocationSizeVirt in qcow2util

Signed-off-by: Damien Thenot <damien.thenot@vates.tech>
---
 drivers/qcow2util.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/qcow2util.py b/drivers/qcow2util.py
index 067907e4..bcb3bba5 100644
--- a/drivers/qcow2util.py
+++ b/drivers/qcow2util.py
@@ -419,6 +419,10 @@ class QCowUtil(CowUtil):
     def getFooterSize(self, path: str) -> int:
         return 0
 
+    @override
+    def getDefaultPreallocationSizeVirt(self) -> int:
+        return 0
+
     @override
     def getMaxChainLength(self) -> int:
         return MAX_QCOW_CHAIN_LENGTH
