From c1823158a04c28e07e7a800c5dcc6fdc1a4ba2c2 Mon Sep 17 00:00:00 2001
From: Ronan Abhamon <ronan.abhamon@vates.tech>
Date: Fri, 26 Sep 2025 18:40:25 +0200
Subject: [PATCH] fix(linstorvolumemanager): report unexpected DRBD device path
 (#100)

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.tech>
---
 drivers/linstorvolumemanager.py | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/linstorvolumemanager.py b/drivers/linstorvolumemanager.py
index 3e339895..c8e89483 100755
--- a/drivers/linstorvolumemanager.py
+++ b/drivers/linstorvolumemanager.py
@@ -991,9 +991,10 @@ class LinstorVolumeManager(object):
         # - "../xcp-volume-<UUID>/0"
         if device_path.startswith(DRBD_BY_RES_PATH):
             prefix_len = len(DRBD_BY_RES_PATH)
-        else:
-            assert device_path.startswith('../')
+        elif device_path.startswith('../'):
             prefix_len = 3
+        else:
+            raise LinstorVolumeManagerError('Unexpected device path: `{}`'.format(device_path))
 
         res_name_end = device_path.find('/', prefix_len)
         assert res_name_end != -1
