From a5a1571f3082fecd270705922b4dd127d3b559ff Mon Sep 17 00:00:00 2001
From: Ronan Abhamon <ronan.abhamon@vates.tech>
Date: Tue, 3 Mar 2026 00:25:27 +0100
Subject: [PATCH] fix(linstor): set `vdiType` on info object

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.tech>
---
 drivers/cleanup.py | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/cleanup.py b/drivers/cleanup.py
index 2384fb99..1a9fed10 100755
--- a/drivers/cleanup.py
+++ b/drivers/cleanup.py
@@ -3553,9 +3553,11 @@ class LinstorSR(SR):
         def load_info(vdi, multi_cowutil):
             vdi_uuid, vdi_type = vdi
             try:
-                return multi_cowutil.get_local_cowutil(vdi_type).get_info(vdi_uuid)
+                vdiInfo = multi_cowutil.get_local_cowutil(vdi_type).get_info(vdi_uuid)
             except Exception as e:
-                return handle_fail(vdi_uuid, e)
+                vdiInfo = handle_fail(vdi_uuid, e)
+            vdiInfo.vdiType = vdi_type
+            return vdiInfo
 
         try:
             for vdiInfo in multi_cowutil.run(load_info, pending_vdis):
