From 02b497c783fab888e8127ab5d1dd8b58632d2995 Mon Sep 17 00:00:00 2001
From: Ronan Abhamon <ronan.abhamon@vates.fr>
Date: Thu, 17 Aug 2023 14:52:13 +0200
Subject: [PATCH 128/178] fix(LinstorSR): mare sure hostnames are unique at SR
 creation

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>
---
 drivers/LinstorSR.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/LinstorSR.py b/drivers/LinstorSR.py
index 98919a4..e512487 100755
--- a/drivers/LinstorSR.py
+++ b/drivers/LinstorSR.py
@@ -649,6 +649,12 @@ class LinstorSR(SR.SR):
             hostname = record['hostname']
             ips[hostname] = record['address']
 
+        if len(ips) != len(online_hosts):
+            raise xs_errors.XenError(
+                'LinstorSRCreate',
+                opterr='Multiple hosts with same hostname'
+            )
+
         # Ensure ports are opened and LINSTOR satellites
         # are activated. In the same time the drbd-reactor instances
         # must be stopped.
-- 
2.46.0

