From 54bca8660c18628ec36c1835b9f14aec80c187a8 Mon Sep 17 00:00:00 2001
From: Ronan Abhamon <ronan.abhamon@vates.fr>
Date: Fri, 18 Aug 2023 11:06:56 +0200
Subject: [PATCH 128/178] fix(LinstorSR): ensure we can attach non-special
 static VDIs

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>
---
 drivers/LinstorSR.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/LinstorSR.py b/drivers/LinstorSR.py
index e512487..1708370 100755
--- a/drivers/LinstorSR.py
+++ b/drivers/LinstorSR.py
@@ -461,6 +461,9 @@ class LinstorSR(SR.SR):
                             logger=util.SMlog,
                             attempt_count=attempt_count
                         )
+                        # Only required if we are attaching from config using a non-special VDI.
+                        # I.e. not an HA volume.
+                        self._vhdutil = LinstorVhdUtil(self.session, self._linstor)
 
                     controller_uri = get_controller_uri()
                     if controller_uri:
