From edbda3949f79cf1fa7d76f4352b7ce20e8161999 Mon Sep 17 00:00:00 2001
From: Damien Thenot <damien.thenot@vates.tech>
Date: Mon, 26 May 2025 17:54:33 +0200
Subject: [PATCH] Add is_shared function on SR

Signed-off-by: Damien Thenot <damien.thenot@vates.tech>
---
 drivers/SR.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/SR.py b/drivers/SR.py
index d9f25197..22bf5051 100755
--- a/drivers/SR.py
+++ b/drivers/SR.py
@@ -152,6 +152,7 @@ class SR(object):
         self.direct = False
         self.ops_exclusive = []
         self.driver_config = {}
+        self._is_shared = None
 
         self.load(sr_uuid)
 
@@ -233,6 +234,11 @@ class SR(object):
         except:
             pass
 
+    def is_shared(self):
+        if not self._is_shared:
+            self._is_shared = self.session.xenapi.SR.get_shared(self.sr_ref)
+        return self._is_shared
+
     def create(self, uuid, size) -> None:
         """Create this repository.
         This operation may delete existing data.
