From a436ed7e607bf5a46db022076ae4fbad243e810f Mon Sep 17 00:00:00 2001
From: Damien Thenot <damien.thenot@vates.tech>
Date: Mon, 2 Jun 2025 14:13:13 +0200
Subject: [PATCH] fix(blktap2.py): change chain check position to be after
 lvchange lock

Signed-off-by: Damien Thenot <damien.thenot@vates.tech>
---
 drivers/blktap2.py | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/blktap2.py b/drivers/blktap2.py
index cb07a07c..f6537d18 100755
--- a/drivers/blktap2.py
+++ b/drivers/blktap2.py
@@ -1770,8 +1770,6 @@ class VDI(object):
 
             vdi_type = self.target.get_vdi_type()
 
-            self._check_journal_coalesce_chain(sr_uuid, vdi_uuid)
-            #TODO: handling error here
 
             # Take lvchange-p Lock before running
             # tap-ctl open
@@ -1784,6 +1782,9 @@ class VDI(object):
                 lock = Lock("lvchange-p", NS_PREFIX_LVM + sr_uuid)
                 lock.acquire()
 
+            self._check_journal_coalesce_chain(sr_uuid, vdi_uuid)
+            #TODO: handling error here
+
             # When we attach a static VDI for HA, we cannot communicate with
             # xapi, because has not started yet. These VDIs are raw.
             if VdiType.isCowImage(vdi_type):
