From 508980797ff8ee27c3a4cfee6d3694988fb0b31d Mon Sep 17 00:00:00 2001
From: Damien Thenot <damien.thenot@vates.tech>
Date: Mon, 6 Oct 2025 17:07:11 +0200
Subject: [PATCH] fix(qcow2): Make getDefaultPreallocationSizeVirt return
 minsize

Signed-off-by: Damien Thenot <damien.thenot@vates.tech>
---
 drivers/qcow2util.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/qcow2util.py b/drivers/qcow2util.py
index 7d08edde..0e254f27 100755
--- a/drivers/qcow2util.py
+++ b/drivers/qcow2util.py
@@ -450,7 +450,8 @@ class QCowUtil(CowUtil):
 
     @override
     def getDefaultPreallocationSizeVirt(self) -> int:
-        return MAX_QCOW_SIZE
+        """vhdutil answer max size (2TiB) here but we don't want to allocate for max size in QCOW2, it would make small LV a lot bigger."""
+        return MIN_QCOW_SIZE
 
     @override
     def getMaxChainLength(self) -> int:
