From 3521b4adce7a1c9453ab5aecc2b27cbbfad53e0f Mon Sep 17 00:00:00 2001
From: Damien Thenot <damien.thenot@vates.tech>
Date: Mon, 2 Jun 2025 16:13:11 +0200
Subject: [PATCH] fix(on-slave): temporary fix for linstor import error

Signed-off-by: Damien Thenot <damien.thenot@vates.tech>
---
 drivers/on_slave.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/on_slave.py b/drivers/on_slave.py
index 0fc4f499..4d98c550 100755
--- a/drivers/on_slave.py
+++ b/drivers/on_slave.py
@@ -83,7 +83,6 @@ def _is_open(session, args):
     import EXTSR
     import LargeBlockSR
     import GlusterFSSR
-    import LinstorSR
     import LVMSR
     import MooseFSSR
     import NFSSR
