From 75b730eb5d79d86b11bf85d527addb599fb0544e Mon Sep 17 00:00:00 2001
From: Damien Thenot <damien.thenot@vates.tech>
Date: Wed, 24 Sep 2025 17:15:41 +0200
Subject: [PATCH] Mock check_journal_coalesce for VDI activation

Signed-off-by: Damien Thenot <damien.thenot@vates.tech>
---
 tests/test_blktap2.py | 18 ++++++++++++------
 1 file changed, 12 insertions(+), 6 deletions(-)

diff --git a/tests/test_blktap2.py b/tests/test_blktap2.py
index c791c9ae..791c8521 100644
--- a/tests/test_blktap2.py
+++ b/tests/test_blktap2.py
@@ -259,8 +259,9 @@ class TestVDI(unittest.TestCase):
     @mock.patch('blktap2.VDI.BackendLink', autospec=True)
     @mock.patch('blktap2.VDI.NBDLink', autospec=True)
     @mock.patch('blktap2.Tapdisk')
+    @mock.patch('blktap2.VDI._check_journal_coalesce_chain', autospec=True)
     def test_activate_relink_retry(
-            self, mock_tapdisk, mock_nbd_link, mock_backend,
+            self, mock_checkjournalcoalesce, mock_tapdisk, mock_nbd_link, mock_backend,
             mock_attach, mock_this_host, mock_sleep):
         """
         Test blktap2.VDI.activate, relinking, retry 1, success
@@ -285,8 +286,9 @@ class TestVDI(unittest.TestCase):
     @mock.patch('blktap2.VDI.BackendLink', autospec=True)
     @mock.patch('blktap2.VDI.NBDLink', autospec=True)
     @mock.patch('blktap2.Tapdisk')
+    @mock.patch('blktap2.VDI._check_journal_coalesce_chain', autospec=True)
     def test_activate_pause_retry(
-            self, mock_tapdisk, mock_nbd_link, mock_backend,
+            self, mock_checkjournalcoalesce, mock_tapdisk, mock_nbd_link, mock_backend,
             mock_attach, mock_this_host, mock_sleep):
         """
         Test blktap2.VDI.activate, paused, retry 1, success
@@ -310,8 +312,9 @@ class TestVDI(unittest.TestCase):
     @mock.patch('blktap2.VDI.BackendLink', autospec=True)
     @mock.patch('blktap2.VDI.NBDLink', autospec=True)
     @mock.patch('blktap2.Tapdisk')
+    @mock.patch('blktap2.VDI._check_journal_coalesce_chain', autospec=True)
     def test_activate_paused_while_tagging(
-            self, mock_tapdisk, mock_nbd_link, mock_backend,
+            self, mock_checkjournalcoalesce, mock_tapdisk, mock_nbd_link, mock_backend,
             mock_attach, mock_this_host, mock_sleep):
         """
         Test blktap2.VDI.activate, paused, while tagging, success
@@ -341,8 +344,9 @@ class TestVDI(unittest.TestCase):
     @mock.patch('blktap2.VDI.BackendLink', autospec=True)
     @mock.patch('blktap2.VDI.NBDLink', autospec=True)
     @mock.patch('blktap2.Tapdisk')
+    @mock.patch('blktap2.VDI._check_journal_coalesce_chain', autospec=True)
     def test_activate_relink_while_tagging(
-            self, mock_tapdisk, mock_nbd_link, mock_backend,
+            self, mock_checkjournalcoalesce, mock_tapdisk, mock_nbd_link, mock_backend,
             mock_attach, mock_this_host, mock_sleep):
         """
         Test blktap2.VDI.activate, relinking, while tagging, retry 1, success
@@ -372,8 +376,9 @@ class TestVDI(unittest.TestCase):
     @mock.patch('blktap2.VDI.BackendLink', autospec=True)
     @mock.patch('blktap2.VDI.NBDLink', autospec=True)
     @mock.patch('blktap2.Tapdisk')
+    @mock.patch('blktap2.VDI._check_journal_coalesce_chain', autospec=True)
     def test_activate_ro_already_activating_retry(
-            self, mock_tapdisk, mock_nbd_link, mock_backend,
+            self, mock_checkjournalcoalesce, mock_tapdisk, mock_nbd_link, mock_backend,
             mock_attach, mock_this_host, mock_sleep):
         """
         If we're activating for read-only access, with someone else (let's
@@ -408,8 +413,9 @@ class TestVDI(unittest.TestCase):
     @mock.patch('blktap2.VDI.BackendLink', autospec=True)
     @mock.patch('blktap2.VDI.NBDLink', autospec=True)
     @mock.patch('blktap2.Tapdisk')
+    @mock.patch('blktap2.VDI._check_journal_coalesce_chain', autospec=True)
     def test_activate_rw_already_activating_fail(
-            self, mock_tapdisk, mock_nbd_link, mock_backend,
+            self, mock_checkjournalcoalesce, mock_tapdisk, mock_nbd_link, mock_backend,
             mock_attach, mock_this_host, mock_sleep):
         """
         If we're activating for read-write access, with someone else (let's
