From 8d930fa4fe31250c3323106cdbd9e832cd9d0343 Mon Sep 17 00:00:00 2001
From: Damien Thenot <damien.thenot@vates.tech>
Date: Fri, 26 Sep 2025 18:11:46 +0200
Subject: [PATCH] qcow2util: read commit size correctly

Signed-off-by: Damien Thenot <damien.thenot@vates.tech>
---
 drivers/qcow2util.py | 15 ++++++++++++---
 1 file changed, 12 insertions(+), 3 deletions(-)

diff --git a/drivers/qcow2util.py b/drivers/qcow2util.py
index 423fae28..99113246 100644
--- a/drivers/qcow2util.py
+++ b/drivers/qcow2util.py
@@ -758,11 +758,20 @@ class QCowUtil(CowUtil):
 
     @override
     def coalesce(self, path: str) -> int:
-            allocated_blocks = self.getAllocatedSize(path)
             # -d on commit make it not empty the original image since we don't intend to keep it
             cmd = [QEMU_IMG, "commit", "-f", QCOW2_TYPE, path, "-d"]
-            ret = cast(str, self._ioretry(cmd)) #TODO: parse for byte coalesced, our qemu-img is supposed to be patched to output it.
-            return allocated_blocks
+            ret = cast(str, self._ioretry(cmd)) # Allows to parse for byte coalesced, our qemu-img is supposed to be patched to output it.
+            lines = ret.splitlines()
+            if re.match("Image committed.", lines[-1]):
+                res_line = lines[-2]
+            else:
+                res_line = lines[-1]
+
+            results = re.match(r"\((\d+)/(\d+)\)", res_line)
+            if results:
+                committed_bytes = int(results.group(1))
+                return committed_bytes
+            raise xs_errors.XenError("TapdiskFailed", "Couldn't get commited size from qemu-img commit call") # TODO: We might not want to raise in this case, it would break if the qemu-img called isn't modified to print the coalesce result even if it succeeded in coalesceing
 
     @override
     def create(self, path: str, size: int, static: bool, msize: int = 0, block_size: Optional[int] = None) -> None:
