From dfcec4b53eaaca42548c5e2d6582404b68817a6c Mon Sep 17 00:00:00 2001
From: Ronan Abhamon <ronan.abhamon@vates.fr>
Date: Mon, 12 Feb 2024 20:54:06 +0100
Subject: [PATCH 164/175] fix(LinstorSR): activate VG if attach from config is
 asked

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>
---
 drivers/LinstorSR.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/LinstorSR.py b/drivers/LinstorSR.py
index de10f42..b6b1529 100755
--- a/drivers/LinstorSR.py
+++ b/drivers/LinstorSR.py
@@ -382,7 +382,7 @@ class LinstorSR(SR.SR):
 
         def load(self, *args, **kwargs):
             # Activate all LVMs to make drbd-reactor happy.
-            if self.srcmd.cmd == 'sr_attach':
+            if self.srcmd.cmd in ('sr_attach', 'vdi_attach_from_config'):
                 activate_lvm_group(self._group_name)
 
             if not self._has_session:
-- 
2.45.2

