From 5491fbc0f245095e62f83b78e3afcd76745cbe5e Mon Sep 17 00:00:00 2001
From: Ronan Abhamon <ronan.abhamon@vates.fr>
Date: Mon, 15 Apr 2024 11:26:00 +0200
Subject: [PATCH 168/180] fix(linstorvolumemanager): blocks deletion of default
 network interface

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>
---
 drivers/linstorvolumemanager.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/linstorvolumemanager.py b/drivers/linstorvolumemanager.py
index 46c3283..02059d8 100755
--- a/drivers/linstorvolumemanager.py
+++ b/drivers/linstorvolumemanager.py
@@ -1491,6 +1491,12 @@ class LinstorVolumeManager(object):
         :param str node_name: Node name of the interface to remove.
         :param str name: Interface to remove.
         """
+
+        if name == 'default':
+            raise LinstorVolumeManagerError(
+                'Unable to delete the default interface of a node!'
+            )
+
         result = self._linstor.netinterface_delete(node_name, name)
         errors = self._filter_errors(result)
         if errors:
-- 
2.46.0

