From 84e236ff9a967f5df25535213f89e377d51d4b5f Mon Sep 17 00:00:00 2001
From: Ronan Abhamon <ronan.abhamon@vates.fr>
Date: Wed, 24 Apr 2024 15:10:49 +0200
Subject: [PATCH 170/180] fix(LinstorSR): fix bad exception reference during
 snapshot

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>
---
 drivers/LinstorSR.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/LinstorSR.py b/drivers/LinstorSR.py
index 3421f79f..3bd31e9e 100755
--- a/drivers/LinstorSR.py
+++ b/drivers/LinstorSR.py
@@ -2500,17 +2500,17 @@ class LinstorVDI(VDI.VDI):
             self.session.xenapi.VDI.set_sm_config(
                 vdi_ref, active_vdi.sm_config
             )
-        except Exception:
+        except Exception as e:
             util.logException('Failed to snapshot!')
             try:
                 self.sr._handle_interrupted_clone(
                     active_uuid, clone_info, force_undo=True
                 )
                 self.sr._journaler.remove(LinstorJournaler.CLONE, active_uuid)
-            except Exception as e:
+            except Exception as clean_error:
                 util.SMlog(
                     'WARNING: Failed to clean up failed snapshot: {}'
-                    .format(e)
+                    .format(clean_error)
                 )
             raise xs_errors.XenError('VDIClone', opterr=str(e))
 
