diff --git i/drivers/NFSSR.py w/drivers/NFSSR.py
index 6590640..1a028ca 100755
--- i/drivers/NFSSR.py
+++ w/drivers/NFSSR.py
@@ -116,11 +116,11 @@ class NFSSR(FileSR.SharedFileSR):
     def check_server(self):
         try:
             if PROBEVERSION in self.dconf:
-                sv = nfs.get_supported_nfs_versions(self.remoteserver)
+                sv = nfs.get_supported_nfs_versions(self.remoteserver, self.transport)
                 if len(sv):
                     self.nfsversion = sv[0]
             else:
-                nfs.check_server_tcp(self.remoteserver, self.nfsversion)
+                nfs.check_server_tcp(self.remoteserver, self.transport, self.nfsversion)
         except nfs.NfsException as exc:
             raise xs_errors.XenError('NFSVersion',
                                      opterr=exc.errstr)
@@ -168,7 +168,7 @@ class NFSSR(FileSR.SharedFileSR):
 
         self.mount(temppath, self.remotepath)
         try:
-            return nfs.scan_srlist(temppath, self.dconf)
+            return nfs.scan_srlist(temppath, self.transport, self.dconf)
         finally:
             try:
                 nfs.unmount(temppath, True)
@@ -255,7 +255,7 @@ class NFSSR(FileSR.SharedFileSR):
 
     def scan_exports(self, target):
         util.SMlog("scanning2 (target=%s)" % target)
-        dom = nfs.scan_exports(target)
+        dom = nfs.scan_exports(target, self.transport)
         print(dom.toprettyxml(), file=sys.stderr)
 
     def set_transport(self):
diff --git i/drivers/nfs.py w/drivers/nfs.py
index 3bed7ad..28fc9f8 100644
--- i/drivers/nfs.py
+++ w/drivers/nfs.py
@@ -43,6 +43,8 @@ import time
 
 RPCINFO_BIN = "/usr/sbin/rpcinfo"
 SHOWMOUNT_BIN = "/usr/sbin/showmount"
+NFS_STAT = "/usr/sbin/nfsstat"
+LS = "/usr/bin/ls"
 
 DEFAULT_NFSVERSION = '3'
 
@@ -52,6 +54,8 @@ NFS_VERSION = [
 NFS_SERVICE_WAIT = 30
 NFS_SERVICE_RETRY = 6
 
+NSFv4_PSEUDOFS = "/"
+NFS4_TMP_MOUNTPOINT = "/tmp/mnt"
 
 class NfsException(Exception):
 
@@ -59,26 +63,30 @@ class NfsException(Exception):
         self.errstr = errstr
 
 
-def check_server_tcp(server, nfsversion=DEFAULT_NFSVERSION):
+def check_server_tcp(server, transport, nfsversion=DEFAULT_NFSVERSION):
     """Make sure that NFS over TCP/IP V3 is supported on the server.
 
     Returns True if everything is OK
     False otherwise.
     """
     try:
-        sv = get_supported_nfs_versions(server)
+        sv = get_supported_nfs_versions(server, transport)
         return (True if nfsversion in sv else False)
     except util.CommandException as inst:
         raise NfsException("rpcinfo failed or timed out: return code %d" %
                            inst.code)
 
 
-def check_server_service(server):
+def check_server_service(server, transport):
     """Ensure NFS service is up and available on the remote server.
 
     Returns False if fails to detect service after 
     NFS_SERVICE_RETRY * NFS_SERVICE_WAIT
     """
+    
+    sv = get_supported_nfs_versions(server, transport)
+    if sv == ['4']:
+        return True
 
     retries = 0
     errlist = [errno.EPERM, errno.EPIPE, errno.EIO]
@@ -134,7 +142,7 @@ def soft_mount(mountpoint, remoteserver, remotepath, transport, useroptions='',
 
     # Wait for NFS service to be available
     try:
-        if not check_server_service(remoteserver):
+        if not check_server_service(remoteserver, transport):
             raise util.CommandException(
                 code=errno.EOPNOTSUPP,
                 reason='No NFS service on server: `%s`' % remoteserver
@@ -188,43 +196,78 @@ def unmount(mountpoint, rmmountpoint):
             raise NfsException("rmdir failed with error '%s'" % inst.strerror)
 
 
-def scan_exports(target):
+def scan_exports(target, transport):
     """Scan target and return an XML DOM with target, path and accesslist."""
     util.SMlog("scanning")
-    cmd = [SHOWMOUNT_BIN, "--no-headers", "-e", target]
     dom = xml.dom.minidom.Document()
     element = dom.createElement("nfs-exports")
     dom.appendChild(element)
-    for val in util.pread2(cmd).split('\n'):
-        if not len(val):
-            continue
-        entry = dom.createElement('Export')
-        element.appendChild(entry)
-
-        subentry = dom.createElement("Target")
-        entry.appendChild(subentry)
-        textnode = dom.createTextNode(target)
-        subentry.appendChild(textnode)
-
-        # Access is not always provided by showmount return
-        # If none is provided we need to assume "*"
-        array = val.split()
-        path = array[0]
-        access = array[1] if len(array) >= 2 else "*"
-        subentry = dom.createElement("Path")
-        entry.appendChild(subentry)
-        textnode = dom.createTextNode(path)
-        subentry.appendChild(textnode)
-
-        subentry = dom.createElement("Accesslist")
-        entry.appendChild(subentry)
-        textnode = dom.createTextNode(access)
-        subentry.appendChild(textnode)
+    fail = True
+    try:
+        cmd = [SHOWMOUNT_BIN, "--no-headers", "-e", target]
+        for val in util.pread2(cmd).split('\n'):
+            if not len(val):
+                continue
+            entry = dom.createElement('Export')
+            element.appendChild(entry)
+
+            subentry = dom.createElement("Target")
+            entry.appendChild(subentry)
+            textnode = dom.createTextNode(target)
+            subentry.appendChild(textnode)
+
+            # Access is not always provided by showmount return
+            # If none is provided we need to assume "*"
+            array = val.split()
+            path = array[0]
+            access = array[1] if len(array) >= 2 else "*"
+            subentry = dom.createElement("Path")
+            entry.appendChild(subentry)
+            textnode = dom.createTextNode(path)
+            subentry.appendChild(textnode)
+
+            subentry = dom.createElement("Accesslist")
+            entry.appendChild(subentry)
+            textnode = dom.createTextNode(access)
+            subentry.appendChild(textnode)
+        fail = False
+    except Exception:
+        util.SMlog("Unable to scan exports with rpcingo, trying NFSv4")
 
+    try:
+        mountpoint = "%s/%s" % NFS4_TMP_MOUNTPOINT, target
+        soft_mount(mountpoint, target, NSFv4_PSEUDOFS, transport, nfsversion='4')
+        paths = util.pread2([LS])
+        unmount(mountpoint, NSFv4_PSEUDOFS)
+        for path in paths.split("\n"):
+            entry = dom.createElement('Export')
+            element.appendChild(entry)
+
+            subentry = dom.createElement("Target")
+            entry.appendChild(subentry)
+            textnode = dom.createTextNode(target)
+            subentry.appendChild(textnode)
+            subentry = dom.createElement("Path")
+            entry.appendChild(subentry)
+            textnode = dom.createTextNode(path)
+            subentry.appendChild(textnode)
+
+            subentry = dom.createElement("Accesslist")
+            entry.appendChild(subentry)
+            # Assume everyone as we do not have any info about it
+            textnode = dom.createTextNode("*")
+            subentry.appendChild(textnode)
+        fail = False
+    except Exception:
+        util.SMlog("Unable to scan exports with NFSv4 pseudo FS mount")
+
+    if fail:
+        raise NfsException('Failed to read NFS export paths from server %s' %
+                           (target))
     return dom
 
 
-def scan_srlist(path, dconf):
+def scan_srlist(path, transport, dconf):
     """Scan and report SR, UUID."""
     dom = xml.dom.minidom.Document()
     element = dom.createElement("SRlist")
@@ -247,7 +290,7 @@ def scan_srlist(path, dconf):
     if PROBEVERSION in dconf:
         util.SMlog("Add supported nfs versions to sr-probe")
         try:
-            supported_versions = get_supported_nfs_versions(dconf.get('server'))
+            supported_versions = get_supported_nfs_versions(dconf.get('server'), transport)
             supp_ver = dom.createElement("SupportedVersions")
             element.appendChild(supp_ver)
 
@@ -263,10 +306,11 @@ def scan_srlist(path, dconf):
     return dom.toprettyxml()
 
 
-def get_supported_nfs_versions(server):
+def get_supported_nfs_versions(server, transport):
     """Return list of supported nfs versions."""
     valid_versions = set(['3', '4'])
     cv = set()
+    res = None
     try:
         ns = util.pread2([RPCINFO_BIN, "-s", "%s" % server])
         ns = ns.split("\n")
@@ -275,12 +319,26 @@ def get_supported_nfs_versions(server):
                 cvi = ns[i].split()[1].split(",")
                 for j in range(len(cvi)):
                     cv.add(cvi[j])
-        return list(cv & valid_versions)
-    except:
-        util.SMlog("Unable to obtain list of valid nfs versions")
+        res = list(cv & valid_versions)
+    except Exception:
+        util.SMlog("Unable to obtain list of valid nfs versions with rpcinfo, trying NSFv4")
+
+    try:
+        mountpoint = "%s/%s" % NFS4_TMP_MOUNTPOINT, server
+        soft_mount(mountpoint, server, NSFv4_PSEUDOFS, transport, nfsversion='4')
+        util.pread2([NFS_STAT, '-m'])
+        unmount(mountpoint, NSFv4_PSEUDOFS)
+        if res is None:
+            res = ['4']
+        else:
+            res.add('4')
+    except Exception:
+        util.SMlog("Unable to obtain list of valid nfs versions with NSFv4 pseudo FS mount")
+
+    if res is None:
         raise NfsException('Failed to read supported NFS version from server %s' %
                            (server))
-
+    return res
 
 def get_nfs_timeout(other_config):
     nfs_timeout = 100
diff --git i/tests/test_NFSSR.py w/tests/test_NFSSR.py
index 29306cc..6762912 100644
--- i/tests/test_NFSSR.py
+++ w/tests/test_NFSSR.py
@@ -121,7 +121,7 @@ class TestNFSSR(unittest.TestCase):
 
         nfssr.attach(None)
 
-        check_server_tcp.assert_called_once_with('aServer',
+        check_server_tcp.assert_called_once_with('aServer', 'tcp',
                                                  'aNfsversionChanged')
         soft_mount.assert_called_once_with('/var/run/sr-mount/UUID',
                                            'aServer',
diff --git i/tests/test_nfs.py w/tests/test_nfs.py
index fd0f257..7d09d0a 100644
--- i/tests/test_nfs.py
+++ w/tests/test_nfs.py
@@ -34,7 +34,7 @@ class Test_nfs(unittest.TestCase):
     @mock.patch('util.pread')
     def test_check_server_service(self, pread, sleep):
         pread.side_effect = ["    100003  4,3,2     udp6,tcp6,udp,tcp                nfs         superuser"]
-        service_found = nfs.check_server_service('aServer')
+        service_found = nfs.check_server_service('aServer', 'tcp')
 
         self.assertTrue(service_found)
         self.assertEqual(len(pread.mock_calls), 1)
@@ -49,7 +49,7 @@ class Test_nfs(unittest.TestCase):
         pread.side_effect = ["",
                            "",
                            "    100003  4,3,2     udp6,tcp6,udp,tcp                nfs         superuser"]
-        service_found = nfs.check_server_service('aServer')
+        service_found = nfs.check_server_service('aServer', 'tcp')
 
         self.assertTrue(service_found)
         self.assertEqual(len(pread.mock_calls), 3)
@@ -60,7 +60,7 @@ class Test_nfs(unittest.TestCase):
     def test_check_server_service_not_available(self, pread, sleep):
         pread.return_value = ""
 
-        service_found = nfs.check_server_service('aServer')
+        service_found = nfs.check_server_service('aServer', 'tcp')
 
         self.assertFalse(service_found)
 
@@ -70,7 +70,7 @@ class Test_nfs(unittest.TestCase):
     def test_check_server_service_exception(self, pread, sleep):
         pread.side_effect = [util.CommandException(errno.ENOMEM)]
         with self.assertRaises(util.CommandException):
-            nfs.check_server_service('aServer')
+            nfs.check_server_service('aServer', 'tcp')
 
     @mock.patch('time.sleep', autospec=True)
     # Can't use autospec due to http://bugs.python.org/issue17826
@@ -78,7 +78,7 @@ class Test_nfs(unittest.TestCase):
     def test_check_server_service_first_call_exception(self, pread, sleep):
         pread.side_effect = [util.CommandException(errno.EPIPE),
                             "    100003  4,3,2     udp6,tcp6,udp,tcp                nfs         superuser"]
-        service_found = nfs.check_server_service('aServer')
+        service_found = nfs.check_server_service('aServer', 'tcp')
 
         self.assertTrue(service_found)
         self.assertEqual(len(pread.mock_calls), 2)
@@ -106,7 +106,7 @@ class Test_nfs(unittest.TestCase):
         nfs.soft_mount('mountpoint', 'remoteserver', 'remotepath', 'transport',
                        timeout=None)
 
-        check_server_service.assert_called_once_with('remoteserver')
+        check_server_service.assert_called_once_with('remoteserver', 'transport')
         pread.assert_called_once_with(self.get_soft_mount_pread('mount.nfs',
                                                                 '3'))
 
@@ -117,7 +117,7 @@ class Test_nfs(unittest.TestCase):
         nfs.soft_mount('mountpoint', 'remoteserver', 'remotepath', 'tcp6',
                        timeout=None)
 
-        check_server_service.assert_called_once_with('remoteserver')
+        check_server_service.assert_called_once_with('remoteserver', 'transport')
         pread.assert_called_once_with(self.get_soft_mount_pread('mount.nfs',
                                                                 '3', True))
 
@@ -129,7 +129,7 @@ class Test_nfs(unittest.TestCase):
         nfs.soft_mount('mountpoint', 'remoteserver', 'remotepath', 'transport',
                        timeout=None, nfsversion='3')
 
-        check_server_service.assert_called_once_with('remoteserver')
+        check_server_service.assert_called_once_with('remoteserver', 'transport')
         pread.assert_called_with(self.get_soft_mount_pread('mount.nfs',
                                                                 '3'))
 
@@ -141,7 +141,7 @@ class Test_nfs(unittest.TestCase):
         nfs.soft_mount('mountpoint', 'remoteserver', 'remotepath', 'transport',
                        timeout=None, nfsversion='4')
 
-        check_server_service.assert_called_once_with('remoteserver')
+        check_server_service.assert_called_once_with('remoteserver', 'transport')
         pread.assert_called_with(self.get_soft_mount_pread('mount.nfs',
                                                                 '4'))
 