CP-51843: disable read-caching when activating Intellicache

From: Mark Syms <mark.syms@cloud.com>

Signed-off-by: Mark Syms <mark.syms@cloud.com>

diff --git a/drivers/blktap2.py b/drivers/blktap2.py
index d55e0f8..96ae76f 100755
--- a/drivers/blktap2.py
+++ b/drivers/blktap2.py
@@ -1945,7 +1945,7 @@ class VDI(object):
                 raise
 
         secondary = "%s:%s" % (self.target.get_vdi_type(),
-                self.PhyLink.from_uuid(sr_uuid, vdi_uuid).readlink())
+                               (self.PhyLink.from_uuid(sr_uuid, vdi_uuid).readlink()))
 
         util.SMlog("Parent tapdisk: %s" % prt_tapdisk)
         leaf_tapdisk = Tapdisk.find_by_path(local_leaf_path)
@@ -1957,6 +1957,8 @@ class VDI(object):
             child_options["existing_prt"] = prt_tapdisk.minor
             child_options["secondary"] = secondary
             child_options["standby"] = scratch_mode
+            # Disable memory read caching
+            child_options.pop("o_direct", None)
             try:
                 leaf_tapdisk = \
                     Tapdisk.launch_on_tap(blktap, local_leaf_path,
