CP-51843: remove unused params

From: Mark Syms <mark.syms@cloud.com>

diff --git a/drivers/blktap2.py b/drivers/blktap2.py
index a2adfe9..d55e0f8 100755
--- a/drivers/blktap2.py
+++ b/drivers/blktap2.py
@@ -1761,7 +1761,7 @@ class VDI(object):
         else:
             if major == Tapdisk.major():
                 self._tap_deactivate(minor)
-                self.remove_cache(sr_uuid, vdi_uuid, caching_params)
+                self.remove_cache(caching_params)
 
         # Remove the backend link
         back_link.unlink()
@@ -1973,7 +1973,7 @@ class VDI(object):
         self.tap = leaf_tapdisk
         return leaf_tapdisk.get_devpath()
 
-    def remove_cache(self, sr_uuid, vdi_uuid, params):
+    def remove_cache(self, params):
         if not self.target.has_cap("SR_CACHING"):
             return
 
