Index: sm-1.25.0/drivers/cleanup.py
===================================================================
--- sm-1.25.0/drivers/cleanup.py
+++ sm-1.25.0/drivers/cleanup.py	2019-02-07 15:38:21.999908265 +0100
@@ -2535,7 +2535,7 @@
     if type in ["lvm", "lvmoiscsi", "lvmohba", "lvmofcoe"]:
         # temporary while LVHD is symlinked as LVM
         type = SR.TYPE_LVHD
-    if type in ["ext", "nfs", "ocfsoiscsi", "ocfsohba", "smb"]:
+    if type in ["ext", "ext4", "nfs", "ocfsoiscsi", "ocfsohba", "smb", "xfs"]:
         type = SR.TYPE_FILE
     if not type in SR.TYPES:
         raise util.SMException("Unsupported SR type: %s" % type)
